quarkus-extension/engine/runtime/src/main/java/org/camunda/bpm/quarkus/engine/extension/impl/CamundaEngineConfig.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.quarkus.engine.extension.impl;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(phase = ConfigPhase.RUN_TIME, name = "camunda.bpm")
+public class CamundaEngineConfig {
+
+  /**
+   * The Camunda JobExecutor thread pool config. This thread pool is responsible for running
+   * Camunda jobs.
+   */
+  @ConfigItem
+  public ThreadPoolConfig threadPool;
+
+  @ConfigGroup
+  public static class ThreadPoolConfig {
+    /**
+     * The size of the thread pool assigned to the Quarkus-managed JobExecutor.
+     *
+     * Default value is 10.
+     */
+    @ConfigItem(defaultValue = "10")
+    public int size;
+
+    /**
+     * The size of the Quarkus-managed JobExecutor queue.
+     *
+     * Default value is 3.
+     */
+    @ConfigItem(defaultValue = "3")
```

Comment:
How did you determine the defaults for `queueSize` and `size`?

Reply:
I used the defaults that we recommend in the JBoss/WildFly manual installation guide. These are the same ones that are configured by default for JBoss: https://docs.camunda.org/manual/7.15/installation/full/jboss/manual/#adjust-the-configuration. Do you think they should be different?

Reply:
After the discussion we had, I checked if it makes sense to set both values to `-1`, i.e. unlimited. This is recommended by Quarkus for the `queueSize`. The core pool maximum size is calculated by Quarkus to the greater of 8 * the number of available processors and 200 (if the `max-threads` property is not explicitly defined). The minimum, running threads of the core thread pool is 1.

I also ran some tests, where I explicitly set the core thread pool `maxSize` and `queueSize` to be lower than our "managed" thread pool. The tests started timing out when the core thread pool `maxSize` is 1, but otherwise, the lower values between the core thread pool and our "managed" one are used and the tests complete successfully.

However, the tests run in an environment where no other executor tasks are available, so it's possible that a timeout might happen with higher property values in a production environment. As a result, I would err on the safe side, and set our default values to `-1`.

Reply:
Just saw this by chance. Does `-1` queue size mean unlimited aka `Integer.MAX_VALUE`? If that is the case, please don't use it for the queue. We have had various problems with this, such as CAM-11368 and similar tickets.

Reply:
@ThorbenLindhauer, yes, it's unlimited, i.e. `Integer.MAX_VALUE`. Thanks for pointing me to CAM-11368. The description there makes sense, so I would leave the default `queueSize` to 3. However, I still think it makes sense to set the `maxPoolSize` to unlimited. The value there is `Short.MAX_VALUE`.

Reply:
> However, I still think it makes sense to set the maxPoolSize to unlimited.
Why? Can lead to similar issues that people lock up their threads (e.g. blocking requests that don't return) and then end up with thousands of threads as new jobs are created.

Reply:
Ah, okay. My viewpoint was from the other extreme. If the core pool `maxSize`, or our own "managed" pool size is too low, the jobs may start to time out.